### PR TITLE
Julianification

### PR DESCRIPTION
**Pull request by aofarrel**
List of tasks:
- Delete unneeded comment towards the top --> done
- Unique vars IDs: try to mimic Stephanie's version where she splits on “chr” plus the chromosome number/letter she already found, or at the very least don’t redo the logic --> rejected, as repeating the logic is what the CWL does
- At line 202, write_config should be passed in the gds file, probably as a variable instead of a global --> done
- Rework logic to avoid a useless for loop --> rejected, the for loop is actually required, see commented code
- Rethink the exit in the write_config function in the third task --> rejected for same reason as above
- Clarify the print when erroring in the third task, ie don’t say “matching” in the failure case --> done
- Rename the function at line 196 --> done

### CODE REVIEW

File: vcf-to-gds-wf.wdl [hunk 1]
```diff
@@ -155,10 +155,7 @@ task check_gds {
 	input {
 		File gds
 		Array[File] vcfs
-		String gzvcf = basename(sub(gds, "\\.gds$", ".vcf.gz"))
-		String bgzvcf =  basename(sub(gds, "\\.gds$", ".vcf.bgz"))
-		String uncompressed = basename(sub(gds, "\\.gds$", ".vcf"))
-		String bcf = basename(sub(gds, "\\.gds$", ".bcf"))
+		File? sample
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl]: A better name would be sample_list or sample_file, etc, but this isn't a big deal.

File: vcf-to-gds-wf.wdl [hunk 2]
```diff
@@ -190,43 +187,39 @@ task check_gds {
...
 			# write GDS file
 			f.write("\ngds_file ")
-			py_thisGdsSplitOnChr = "~{gds}".split("chr")
-			f.write("'")
-			f.write(vcf_or_gds_with_space(py_thisGdsSplitOnChr)[0])
-			f.write("'")
+			f.write("'" + split_n_space(py_gds.split("chr"))[0] + "'")
+
+			# write the sample file
+			if(~{sample}):
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl]: I don't have experience w/ optional files passed into Python in a task body. Does this test work?

File: vcf-to-gds-wf.wdl [hunk 3]
```diff
@@ -190,43 +187,39 @@ task check_gds {
...
+				f.write("sample_file ")
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl]: This could all be one line

File: vcf-to-gds-wf.wdl [hunk 4]
```diff
@@ -190,43 +187,39 @@ task check_gds {
...
 			f.close()
 
 			# write chromosome number to new file, to be read in bash
 			g = open("chr_number", "a")
 			g.write(str(py_thisChr)) # already str if chrX but python won't complain
-			exit()
 
 		py_vcfarray = ['~{sep="','" vcfs}']
+		py_gds = "~{gds}"
 		# recall that this is a scattered task and we pass in one gds and the entire vcf array
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl]: Could you remove this comment?

File: vcf-to-gds-wf.wdl [hunk 5]
```diff
@@ -190,43 +187,39 @@ task check_gds {
 				py_thisChr = py_splitstring[1][0:1]
 			return [py_thisVcfWithSpace, py_thisChr]
 
-		def write_config(py_file):
+		def write_config(py_vcf, py_gds):
 			f = open("check_gds.config", "a")
 
 			# write VCF file
 			f.write("vcf_file ")
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl:196]: This could be put on the next line!

File: vcf-to-gds-wf.wdl [hunk 6]
```diff
@@ -190,43 +187,39 @@ task check_gds {
...
-			py_thisVcfSplitOnChr = py_file.split("chr")
-			f.write("'")
-			f.write(vcf_or_gds_with_space(py_thisVcfSplitOnChr)[0])
-			f.write("'")
+			f.write("'" + split_n_space(py_vcf.split("chr"))[0] + "'")
 
 			# write GDS file
 			f.write("\ngds_file ")
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl]: This could be put on the next line!

File: vcf-to-gds-wf.wdl [hunk 7]
```diff
@@ -173,7 +170,7 @@ task check_gds {
 		python << CODE
 		import os
 
-		def vcf_or_gds_with_space(py_splitstring):
+		def split_n_space(py_splitstring):
```
Comment (juklucas) [inline vcf-to-gds-wf.wdl:172]: I think a bit of documentation (about what the function returns) would be good here.
Something along the lines of
```
Returns tuple with [file name w/ chromosome name replaced by space, The chromosome name]
for example: test_data_chrX.vcf.gz returns
["test_data_chr .vcf.gz", "X"]```

**Review by juklucas — CHANGES_REQUESTED** [review]
The third task looks awesome...
The reward for winning the pie eating contest is, however, more pie. The second task needs to be as clean as the third task.

I have also added other small comments for your review.

File: vcf-to-gds-wf.wdl (hunk 6 above)
Comment (aofarrel) [reply vcf-to-gds-wf.wdl]: move newline to previous

File: vcf-to-gds-wf.wdl (hunk 1 above)
Comment (aofarrel) [reply vcf-to-gds-wf.wdl]: Leaving as is to keep one word variable consistency

File: vcf-to-gds-wf.wdl (hunk 2 above)
Comment (aofarrel) [reply vcf-to-gds-wf.wdl]: Oops! https://github.com/aofarrel/TOPMed_Pipeline_First_Step_WDL/issues/24

**Pull request merged**